src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2385 @@
+                JSC::throwTypeError(globalObject, scope, "type must be a string"_s);
+                return JSC::JSValue::encode(JSC::JSValue {});
+            }
+            type = typeJSValue.toWTFString(globalObject);
+            RETURN_IF_EXCEPTION(scope, encodedJSValue());
+        }
+
+        if (type == "pkcs1"_s) {
+            // must be RSA
+            const unsigned char* p = reinterpret_cast<const unsigned char*>(data);
+            auto pkey = EvpPKeyPtr(d2i_PublicKey(EVP_PKEY_RSA, nullptr, &p, byteLength));
+            if (!pkey) {
+                // maybe is a private RSA key
+                const unsigned char* p = reinterpret_cast<const unsigned char*>(data);
+                pkey = EvpPKeyPtr(d2i_PrivateKey(EVP_PKEY_RSA, nullptr, &p, byteLength));
+                if (!pkey) {
+                    throwException(globalObject, scope, createTypeError(globalObject, "Invalid PKCS#1"_s));
+                    return JSValue::encode(JSC::jsUndefined());
+                }
+
+                auto pKeyID = EVP_PKEY_id(pkey.get());
+                return KeyObject__createRSAFromPrivate(globalObject, pkey.get(), pKeyID == EVP_PKEY_RSA_PSS ? CryptoAlgorithmIdentifier::RSASSA_PKCS1_v1_5 : CryptoAlgorithmIdentifier::RSAES_PKCS1_v1_5);
+            }
+
+            auto pKeyID = EVP_PKEY_id(pkey.get());
+            auto impl = CryptoKeyRSA::create(pKeyID == EVP_PKEY_RSA_PSS ? CryptoAlgorithmIdentifier::RSASSA_PKCS1_v1_5 : CryptoAlgorithmIdentifier::RSAES_PKCS1_v1_5, CryptoAlgorithmIdentifier::SHA_1, false, CryptoKeyType::Public, WTFMove(pkey), true, CryptoKeyUsageEncrypt);
+            return JSC::JSValue::encode(JSCryptoKey::create(structure, zigGlobalObject, WTFMove(impl)));
+        } else if (type == "spki"_s) {
+            // We use d2i_PUBKEY() to import a public key.
+            const uint8_t* ptr = reinterpret_cast<const uint8_t*>(data);
+            auto pkey = EvpPKeyPtr(d2i_PUBKEY(nullptr, &ptr, byteLength));
+            if (!pkey) {
+                throwException(globalObject, scope, createTypeError(globalObject, "Invalid public key"_s));
+                return JSValue::encode(JSC::jsUndefined());
+            }
+            auto pKeyID = EVP_PKEY_id(pkey.get());
+
+            if (pKeyID == EVP_PKEY_RSA || pKeyID == EVP_PKEY_RSA_PSS) {
+                auto impl = CryptoKeyRSA::create(pKeyID == EVP_PKEY_RSA_PSS ? CryptoAlgorithmIdentifier::RSA_PSS : CryptoAlgorithmIdentifier::RSA_OAEP, CryptoAlgorithmIdentifier::SHA_1, false, CryptoKeyType::Public, WTFMove(pkey), true, CryptoKeyUsageEncrypt);
+                return JSC::JSValue::encode(JSCryptoKey::create(structure, zigGlobalObject, WTFMove(impl)));
+            } else if (pKeyID == EVP_PKEY_ED25519) {
+                auto result = CryptoKeyOKP::importSpki(CryptoAlgorithmIdentifier::Ed25519, CryptoKeyOKP::NamedCurve::Ed25519, Vector<uint8_t>((uint8_t*)data, byteLength), true, CryptoKeyUsageVerify);
+                if (result == nullptr) {
```

Comment:
optional, nitpicky: all these `== nullptr` should be `UNLIKELY(result == nullptr)`

Reply:
Fixed on https://github.com/oven-sh/bun/pull/5940/commits/91ffca6b675e31882c4998b7f757f6f8c5a0e216